packages/esm/core/readme.md

```diff
@@ -0,0 +1,84 @@
+# @esm/core
+
+## How to use?
+
+@esm/core is ready to be consumed right after building it. To build run
+
+`npm run build -w @esm/core`
+
+For the time being, importing directly from esm/core will need to be done from its dist directory. Of course, this will not be a part of the final architecture.
+
+After building you can install @esm/core using npm link or uding a direct file:// link alongside @babylonjs/core
+
+Any function from the webgl engine is used independently using the engineState object you initially create. This state object holds all variables needed for the different functions to work.
+
+To use, for example:
+
+```javascript
+import { Color4 } from "@babylonjs/core/Maths/math.color";
+import { initWebGLEngineState, clear, setViewport, createTexture } from "@esm/core/dist/Engines/WebGL/engine.webgl";
+
+const canvas = document.getElementById("babylon-canvas") as HTMLCanvasElement;
+
+const engineState = initWebGLEngineState(canvas);
+
+setViewport(engineState, {
+    x: 0,
+    y: 0,
+    width: 1,
+    height: 1
+});
+clear(engineState, Color4.FromInts(83, 156, 33, 255), true, true);
+
+createTexture(engineState, "./alien.png", false, false, null);
+```
+
+## Adapters
+
+You can generate an adapter that will be "converted" to a corresponding Engine class. For example, you can generate a ThinEngine using this command:
+
+```javascript
+const engineState = initWebGLEngineState(glContext, {
+         disableWebGL2Support: true
+      });
+      // get adapter
+      const adapter: Engine = augmentEngineState(engineState, {
+        // functions you are using in your experience, imported from the webgl engine
+
+      });

```

Comment:
You are right, this is the one catch here - if the functions are not imported, it is impossible to know what functions are needed to run a certain component, unless you dive into the code of this component, or run your experience and add each function that the engine is missing (which is how i got different projects to work with esm. Took roughly 10 minutes to do that, but it is, of course, not optimal!)
This is what i was trying to solve with the adapter helpers. It is a tree-shakable list of objects that contain the functions needed to run a specific component. so far I did it for the internal texture and effect, but I am planning on expanding this. 
Extensions, for example, are being added as a whole. So if you use setExtension, it will augment the engine's state object, making these functions available for other components.